src/components/ContactsApp.jsx

```diff
@@ -39,17 +48,31 @@ class ContactsApp extends React.Component {
     });
   };
 
+  displayContactForm = () => {
+    this.setState({
+      isCreationFormDisplayed: true
+    });
+  };
+
+  hideContactForm = () => {
+    this.setState({
+      isCreationFormDisplayed: false
+    });
+  };
+
   onDeleteContact = contact => {
     this.props.deleteContact(contact);
     this.hideContactCard();
   };
 
   render() {
-    const { displayedContact } = this.state;
+    const { displayedContact, isCreationFormDisplayed } = this.state;
 
     return (
       <main className="app-content">
-        <ContactsHeaderWithActions />
+        <ContactsHeaderWithActions
+          displayContactForm={this.displayContactForm}
+        />
```

Comment:
Now we understand clearer why I removed some responsibilities to the `<ContactsHeader />` component: it does not have to know anything about the action that displays the contact form.